java-checks/src/main/java/org/sonar/java/checks/tests/AssertJConsecutiveAssertionCheck.java

```diff
@@ -136,26 +140,49 @@ private void reportConsecutiveAssertions(List<StatementTree> statements) {
     return Optional.empty();
   }
 
-  private void reportIssueIfMultipleCalls(@Nullable AssertSubject assertSubject, List<MethodInvocationTree> equivalentAssertions) {
+  private void reportIssueIfMultipleCalls(@Nullable AssertSubject assertSubject, List<AssertSubject> equivalentAssertions) {
     if (assertSubject != null && !equivalentAssertions.isEmpty()) {
-      reportIssue(ExpressionUtils.methodName(assertSubject.mit),
+      reportIssue(assertSubject.methodName(),
         "Join these multiple assertions subject to one assertion chain.",
-        equivalentAssertions.stream().map(mit -> new JavaFileScannerContext.Location("Other assertThat", ExpressionUtils.methodName(mit))).collect(Collectors.toList()),
+        equivalentAssertions.stream().map(AssertSubject::toSecondaryLocation).collect(Collectors.toList()),
         null);
     }
   }
 
   private static class AssertSubject {
     final MethodInvocationTree mit;
+    final Type assertionType;
     final ExpressionTree arg;
 
     AssertSubject(MethodInvocationTree mit, ExpressionTree arg) {
       this.mit = mit;
+      this.assertionType = mit.symbolType().erasure();
       this.arg = arg;
     }
 
     boolean hasEquivalentArgument(AssertSubject other) {
-      return SyntacticEquivalence.areEquivalent(arg, other.arg);
+      return SyntacticEquivalence.areEquivalent(arg, other.arg)
+        && (other.assertionType.isSubtypeOf(assertionType) || couldBeChained(other));
+    }
+
+    boolean couldBeChained(AssertSubject other) {
+      Tree otherParent = other.mit
+        // owning member select
+        .parent()
+        // chained method invocation
+        .parent();
```

Comment:
You can use `MethodTreeUtils.consecutiveMethodInvocation` to do this.

Reply:
Oh nice!